### PR TITLE
Update dependencies, dev-dependencies

### PR DESCRIPTION
**Pull request by wooorm**
This updates the dependencies and dev-dependencies in `packages/`. Unfortunately, either updating to webpack 5 or updating to react 17 crash the webpack loader with a react error, with an [invalid hook call warning](https://reactjs.org/warnings/invalid-hook-call-warning.html) for `useMDXComponents`:

<https://github.com/mdx-js/mdx/blob/dafdf6d70affa5dba0b3b7070f7a310b70bbf775/packages/react/src/context.js#L15>

Which might have to do with the magic of shortcodes (#1385), or something else, I have no clue.

Furthermore, this loosens package dependencies instead of locking them, which relates to GH-865, GH-1015, and GH-1267. It was a long and divided discussion before and the reason for changing now is: While the package currently doesn’t break easily (it was mentioned that unlocking packages might cause that), we are currently *locked* on security vulnerabilities. We’re not getting any patches, and MDX isn’t released that frequently or maintained that actively, so MDX users are stuck. If folks want to lock: npm and yarn have package locks.

Closes GH-1267.
Closes GH-1375.

### CODE REVIEW

File: packages/babel-plugin-apply-mdx-type-props/package.json [hunk 1]
```diff
@@ -25,11 +25,8 @@
     "remark",
     "babel"
   ],
-  "peerDependencies": {
-    "@babel/core": "^7.10.5"
-  },
   "dependencies": {
-    "@babel/helper-plugin-utils": "7.10.4"
+    "@babel/helper-plugin-utils": "^7.10.4"
```
Comment (ChristianMurphy) [inline packages/babel-plugin-apply-mdx-type-props/package.json:29]: would it make sense to broaden these to
```suggestion
    "@babel/helper-plugin-utils": "^7.0.0"
```
?

**Review by ChristianMurphy — APPROVED** [review]
LGTM :+1: 

---

> updating to react 17

In a few other upgrades to React 17, the issue has been transitive dependencies on React 16, causing both React 16 and 17 to be in the tree.
Triggering

> You might have more than one copy of React in the same app

- https://reactjs.org/warnings/invalid-hook-call-warning.html

File: packages/babel-plugin-apply-mdx-type-props/package.json (hunk 1 above)
Comment (wooorm) [reply packages/babel-plugin-apply-mdx-type-props/package.json:29]: I’d say maybe later, I’m not too familiar with the internals there to know if we’re depending on internal features released in those minor versions

**Comment by wooorm** [comment]
> In a few other upgrades to React 17, the issue has been transitive dependencies on React 16, causing both React 16 and 17 to be bundled.

Yeah, that might be it indeed. I’ll try again somewhere soon

**Pull request merged**